cmake/templates/OpenCVConfig.root-WIN32.cmake.in

```diff
@@ -43,10 +43,10 @@ endif()
 
 if(NOT DEFINED OpenCV_STATIC)
   # look for global setting
-  if(NOT DEFINED BUILD_SHARED_LIBS OR BUILD_SHARED_LIBS)
-    set(OpenCV_STATIC OFF)
-  else()
+  if(NOT DEFINED BUILD_SHARED_LIBS OR NOT BUILD_SHARED_LIBS)
```

Comment:
I haven't checked the logic but usually file generated from `*.cmake.in` template file is loaded on a user side where `BUILD_SHARED_LIBS` means user's `BUILD_SHARED_LIBS`, not `BUILD_SHARED_LIBS` for OpenCV (or other third party). Probably this part simply means `set(OpenCV_STATIC @OpenCV_STATIC@)`?

Reply:
It only sets this if OpenCV_STATIC isn't defined so I assumed this is the logic to work out whether it is static or not.
Are you saying it should be configured from the CMakeLists instead?

Reply:
> It only sets this if OpenCV_STATIC isn't defined

Defined by user (!)

From my opinion the logic is weak on OpenCV upstream side, because user have to set variables explicitly instead of relying on reasonable defaults. The `set(OpenCV_STATIC @OpenCV_STATIC@)` will fix the issue but I guess it will break upstream feature when shared+static configuration installed simultaneously (which is [tricky in general](https://cgold.readthedocs.io/en/latest/tutorials/libraries/static-shared.html)).